### DISCUSSION TITLE
dev model question

### DISCUSSION THREAD

**Discussion by quantrpeter** [apache/netbeans#6920] [Q&A] [answered]
Hi All
    Every time I change some code in the module, I have to "ant build" in the root folder, then "ant tryme". It takes over 3 mins which is too long waiting. If I just" ant build" in the module folder", it won't take effect. Then what is the fastest way to debug the module?
thanks 
Peter

**Comment by matthiasblaesing** [comment, marked as answer]
Hi, working on the NetBeans codebase works well with the NetBeans IDE itself. My general approach is:

- checkout the base you want to work on `git checkout ...`
- clean the directory tree `git clean -f -x -d`
- build once the full tree `ant`
- open the modules you want to work on in the IDE
- you can now run the (potentially modified) module using the context menu of the project via "Run" or "Debug"

![image](https://github.com/apache/netbeans/assets/2179736/088e99f5-14d6-4259-aea9-0f0f8e575be7)

**Reply by quantrpeter** [reply]
Thank you Sir !!!
